moveit_core/collision_detection_fcl/include/moveit/collision_detection_fcl/collision_common.h

```diff
@@ -213,47 +236,84 @@ struct FCLGeometry
 };
 
+/** \brief Bundles an \e FCLObject and and a broadphase FCL collision manager. */
 struct FCLManager
 {
   FCLObject object_;
   std::shared_ptr<fcl::BroadPhaseCollisionManagerd> manager_;
 };
 
+/** \brief Callback function used by the FCLManager used for each pair of collision objects to
+*   calculate object distances.
+*
+*   \param o1 First FCL collision object
+*   \param o2 Second FCL collision object
+*   \data General pointer to arbitrary data which is used during the callback
+*   \return True terminates the distance check, false let it continue to the next pair of objects */
 bool collisionCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void* data);
 
+/** \brief Callback function used by the FCLManager used for each pair of collision objects to
+*   calculate collisions and distances.
+*
+*   \param o1 First FCL collision object
+*   \param o2 Second FCL collision object
+*   \data General pointer to arbitrary data which is used during the callback
+*   \return True terminates the collision check, false let it continue to the next pair of objects */
 bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void* data, double& min_dist);
 
+/** \brief Create new FCLGeometry object out of robot link model. */
 FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape, const robot_model::LinkModel* link,
                                             int shape_index);
+
+/** \brief Create new FCLGeometry object out of attached body. */
 FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape, const robot_state::AttachedBody* ab,
                                             int shape_index);
+
+/** \brief Create new FCLGeometry object out of a world object.
+ *
+ *  A world object always consists only of a single shape, therefore we don't need the \e shape_index. */
 FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape, const World::Object* obj);
 
+/** \brief Create new scaled and / or padded FCLGeometry object out of robot link model. */
 FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape, double scale, double padding,
                                             const robot_model::LinkModel* link, int shape_index);
+
+/** \brief Create new scaled and / or padded FCLGeometry object out of an attached body. */
 FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape, double scale, double padding,
                                             const robot_state::AttachedBody* ab, int shape_index);
+
+/** \brief Create new scaled and / or padded FCLGeometry object out of an world object. */
 FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape, double scale, double padding,
                                             const World::Object* obj);
+
+/** \brief Increases the counter of the caches which can possibly trigger the cleaning of the them. */
```

Comment:
From the [definition](https://github.com/ros-planning/moveit/blob/167642b3b5d5484295c93478357f18ef0949027c/moveit_core/collision_detection_fcl/src/collision_common.cpp#L368), this function will force true on bump, which always removes entries where weak pointers have expired. I think the better way of saying this is "which can trigger cleaning of expired entries from them."

Also, a rule of thumb that I've found useful is to keep comments about specific implementation in .cpp files and put comments about the interface and side-effects in the header. So, here, just mention that expired entries are removed, but don't mention weak pointers.